lib/src/skin_tone_overlay.dart

```diff
@@ -0,0 +1,111 @@
+import 'package:emoji_picker_flutter/emoji_picker_flutter.dart';
+import 'package:flutter/material.dart';
+
+/// Overlay for SkinTone
+OverlayEntry buildSkinToneOverlay(
+  BuildContext context,
+  Emoji emoji,
+  double emojiSize,
+  CategoryEmoji? categoryEmoji,
+  int index,
+  int skinToneCount,
+  Config config,
+  double scrollControllerOffset,
+  double tabBarHeight,
+  EmojiPickerUtils utils,
+  OnEmojiSelected onEmojiSelected,
+) {
```

Comment:
I like that you moved out the code that belongs to the SkinToneOverlay to a separate file which makes it easier readable. But I would suggest making it a class instead of global method. Maybe something like `SkinToneOverlayBuilder` would be good name.

Reply:
@Fintasys updated with mixin approach, with this approach we're covering SOLID principles.